fetchai/agents-aea
# AEA-587, AEA-596, AEA-597 - expose exception handling and default routing on aea config

## Proposed changes

This PR lets the developer optionally specify two more fields in aea-config:
- skill exception policy (one of "propagate" (default), "just_log", "log_and_exit")
- default routing (a map from protocol ids to connection ids) which defines a default for outgoing envelopes routing

Both can also be set on AEA/Multiplexer respectively.

WIP: validations, tests

## Fixes

-#1232, #1241

## Types of changes

What types of changes does your code introduce to agents-aea?
_Put an `x` in the boxes that apply_


## Checklist

_Put an `x` in the boxes that apply._

- [x] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [x] I have checked that code coverage does not decrease.
- [x] I have checked that the documentation about the `aea cli` tool works

## Further comments

To test the exception policy:
1. `aea create my_agent && cd my_agent && aea add --local skill fetchai/simple_service_registration:0.2.0 && aea add --local connection fetchai/oef:0.3.0 && aea config set agent.default_connection fetchai/oef:0.3.0`
2. Run oef node locally
3. Place `assert False` in the behaviour code of service registration skill
4. `aea run --connections fetchai/oef:0.3.0`

You should see something like:
<img width="680" alt="Screenshot 2020-05-21 at 21 04 22" src="https://user-images.githubusercontent.com/19314578/82602367-e1016f00-9ba8-11ea-8db9-ae31abfb8e42.png">

Now modify the aea-config:
`aea config set agent.skill_exception_policy just_log`
Run again...you should see something like:

<img width="669" alt="Screenshot 2020-05-21 at 21 02 54" src="https://user-images.githubusercontent.com/19314578/82602496-16a65800-9ba9-11ea-8851-a0a9e174c5fd.png">


Finally, modify the aea-config:
`aea config set agent.skill_exception_policy log_and_exit`
Run again...you should see something like:

<img width="682" alt="Screenshot 2020-05-21 at 21 03 57" src="https://user-images.githubusercontent.com/19314578/82602547-29b92800-9ba9-11ea-8962-e81068829de4.png">
